docs/docker/building-net-docker-images.md

```diff
@@ -0,0 +1,189 @@
+
+- `microsoft/dotnet:<version>-core` : i.e. **microsoft/dotnet:1.0.0-core**, image contains only .NET Core (runtime and libraries) and it is optimized for running [portable .NET Core applications](https://docs.microsoft.com/en-us/dotnet/articles/core/app-types).
+
+- `microsoft/dotnet:<version>-core-deps` : i.e. **microsoft/dotnet:1.0.0-core-deps**, if you wish to run self-contained applications use this image.  Contains the operating system with all of the native dependencies needed bt .NET Core. Can also be used to build a custom copy of .NET core by compiling coreclr and corefx.
+
+Latest versions of each variant:
+
+- `microsoft/dotnet` or `microsoft/dotnet:latest` (aka sdk)
+- `microsoft/dotnet:onbuild`
+- `microsoft/dotnet:core`
+- `microsoft/dotnet-core-deps`
+
+Here is a list of the images after a `docker pull <imagename>` on a development machine to show the various sizes.
+
+```
+REPOSITORY          TAG                  IMAGE ID            SIZE
+microsoft/dotnet    onbuild              19b6a6c4b1db        540.4 MB
+microsoft/dotnet    1.0.0-preview2-sdk   a92c3d9ad0e7        540.4 MB
+microsoft/dotnet    core                 5224a9f2a2aa        253.2 MB
+microsoft/dotnet    1.0.0-core-deps      c981a2eebe0e        166.2 MB
+microsoft/dotnet    core-deps            c981a2eebe0e        166.2 MB
+microsoft/dotnet    latest               03c10abbd08a        540.4 MB
+microsoft/dotnet    1.0.0-core           b8da4a1fd280        253.2 MB
+```
+
+## Prerequisites
+
+- [.NET Core](http://dot.net)
+- Docker Engine : see [Docker Installation page](http://www.docker.com/products/docker)
+- Yeoman tooling for ASP.NET for creating the Web API application
+- Yeoman tools for Docker from Microsoft
+
+The Yeoman tools are installed using npm 
+
+```
+npm install -g yo generator-aspnet generator-docker
+```
+
+> [!NOTE]
+> This sample will be using [Visual Studio Code](http://code.visualstudio.com) for the editor.
+
+## Creating the Web API application
+
+Using the ASP.NET Yeoman generator, open a command or terminal session and type the following.
+```
+yo aspnet
+```
+
+Select **Web API Application** and type **api** for the name of the app ann tap enter.  Once the application is scaffolded, change to the `/api` directory and restore the NuGet dependencies using `dotnet restore`
+
+```
+cd api
+dotnet restore
+```
+
+Test the application using `dotnet run` and browsing to **http://localhost:5000/api/values**
+
+```javascript
+[
+    "value1",
+    "value2"
+]
+```
+
+Use `Ctrl+C` to stop the application.
+
+## Adding Docker support
+Adding Docker support the project is achieved using the Yeoman generator from Microsoft. It currently supports .NET Core, node.js and Golang projects by creating a Dockerfile and scripts that help build and run project inside containers. Visual Studio Code specific files are also added (launch.json, tasks.json) for editor debugging and command pallete support.
```

Comment:
*Adding docker support to the project
* build and run projects or build and run a project?